### PR TITLE
PENG-2159 Expanded permission sets

### PR DESCRIPTION
**Pull request by dusktreader**
Remapped permissions to expand from a view/edit paradigm to individual permissions for create/read/update/delete.

Updated the API with the permission sets.
Updated jobbergate-composed to use the new permission sets.

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

**Review by fschuch — CHANGES_REQUESTED** [review]
Nice work. I left a couple of comments for you.

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ async def job_script_delete(
 async def job_script_get_file(
     id: int = Path(...),
     file_name: str = Path(...),
-    secure_services: SecureService = Depends(secure_services(Permissions.JOB_SCRIPTS_VIEW, commit=False)),
+    secure_services: SecureService = Depends(secure_services(Permissions.JOB_SCRIPTS_READ, commit=False)),
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:273]: Is that the only time when you fetch the job_script file, though? Isn't this just a generic endpoint for fetching the file? I worry that adding the other permission would be too limiting for the purpose of this endpoint.

File: jobbergate-composed/etc/jobbergate-local.json [hunk 2]
```diff
@@ -592,12 +694,18 @@
...
+          "jobbergate:job-scripts:update",
+          "jobbergate:job-scripts:delete",
+          "jobbergate:job-submissions:create",
+          "jobbergate:job-submissions:read",
+          "jobbergate:job-submissions:update",
+          "jobbergate:job-submissions:delete",
+          "jobbergate:job-templates:create",
+          "jobbergate:job-templates:read",
+          "jobbergate:job-templates:update",
+          "jobbergate:job-templates:delete"
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-composed/etc/jobbergate-local.json]: Addressed in commit 87ff2d0.

**Review by fschuch — APPROVED** [review]

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:273]: I think we will never have a situation where someone has permission to submit job scripts but does not have permission to see the job scripts that they want to submit from. They will have to have permission to at least see the job scripts to select one to submit.

**Review by fschuch — APPROVED** [review]

**Pull request merged**